docs/control-structures.rst

```diff
@@ -100,17 +100,17 @@ Functions marked with ``@pure`` cannot call non-``pure`` functions.
 Re-entrancy Locks
 -----------------
 
-The ``@nonreentrant(<key>)`` decorator places a lock on a function, and all functions with the same ``<key>`` value. An attempt by an external contract to call back into any of these functions causes the transaction to revert.
+The ``@nonreentrant`` decorator places a global nonreentrancy lock on a function. An attempt by an external contract to call back into any other ``@nonreentrant`` function causes the transaction to revert.
 
 .. code-block:: python
 
     @external
-    @nonreentrant("lock")
+    @nonreentrant
     def make_a_call(_addr: address):
         # this function is protected from re-entrancy
         ...
 
-You can put the ``@nonreentrant(<key>)`` decorator on a ``__default__`` function but we recommend against it because in most circumstances it will not work in a meaningful way.
+You can put the ``@nonreentrant`` decorator on a ``__default__`` function but we recommend against it because in most circumstances it will not work in a meaningful way.
```

Comment:
> What about disallowing this pattern completely and making instead `__default__` reentrancy-safe by default in the compiler?

i can never remember why we added that note -- i think it's there because often when you have a default function, it's intended to be called back to, so you don't want to add the nonreentrancy lock

Reply:
The underlying reason why I was thinking about this was our intention to disable reentrancy by default (c.f. https://github.com/vyperlang/vyper/issues/3380). So we will need to think about how to handle the special `__default__` case. But we can address this once we get there.